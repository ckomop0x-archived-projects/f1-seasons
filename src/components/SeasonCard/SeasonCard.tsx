import { FC } from 'react';
import CardButton from 'styles/CardButton';

import Flag from '../Flag';

import {
  CircuitName,
  CountryDetails,
  DetailsButton,
  RaceName,
  Round,
  RoundNumber,
  RoundTitle,
  SeasonCardWrapper
} from 'components/SeasonCard/styles';

export interface SeasonCardProps {
  raceName: string;
  circuitName: string;
  season: string;
  round: string;
  country: string;
  locality: string;
  circuitId: string;
  date: string;
  time: string;
  onSeasonSelect(): void;
}

const SeasonCard: FC<SeasonCardProps> = ({
  raceName,
  circuitName,
  round,
  country,
  locality,
  date,
  onSeasonSelect
}) => (
  <SeasonCardWrapper title={circuitName}>
    <Round>
      <RoundTitle>Round</RoundTitle>
      <RoundNumber>{round}</RoundNumber>
    </Round>
    <CountryDetails>
      <Flag country={country} />
      <CircuitName>
        <h3>{circuitName}</h3>
        <span>
          {locality}, {country}
        </span>
      </CircuitName>
    </CountryDetails>

    <RaceName>
      <b>Race:</b> {raceName}
    </RaceName>
    <RaceName>
      <b>Date:</b> {date}
    </RaceName>
    <DetailsButton onClick={onSeasonSelect} title="Go to standings">
      Details
    </DetailsButton>
  </SeasonCardWrapper>
);

export default SeasonCard;
