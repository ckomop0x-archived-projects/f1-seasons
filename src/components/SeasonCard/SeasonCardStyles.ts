import styled from 'styled-components';

export const Card = styled.div`
  width: 100%;
  margin-bottom: ${({ theme }) => theme.grid * 2}px;
  background: white;
  color: ${({ theme }) => theme.gray1};
  box-shadow: 0 0 10px 1px rgba(0, 0, 0, 0.3);

  @media (min-width: 600px) {
    margin: 8px;
    width: 47%;
  }

  @media (min-width: 900px) {
    width: 31%;
  }

  @media (min-width: 1200px) {
    width: 22%;
  }

  @media (min-width: 1500px) {
    width: 18%;
  }

  .content {
    padding: ${({ theme }) => theme.grid * 2.5}px;
    background: ${({ theme }) => theme.gray2};
  }

  h4 {
    text-align: center;
  }
`;
