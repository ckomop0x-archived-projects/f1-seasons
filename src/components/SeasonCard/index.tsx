import { FC } from 'react';
import styled from 'styled-components';

import Flag from '../Flag';

import CardButton from 'styles/CardButton';

export interface SeasonCardProps {
  raceName: string;
  circuitName: string;
  season: string;
  round: string;
  country: string;
  locality: string;
  circuitId: string;
  date: string;
  time: string;
  onSeasonSelect(): void;
}

const SeasonCard: FC<SeasonCardProps> = ({
  raceName,
  circuitName,
  round,
  country,
  locality,
  date,
  onSeasonSelect
}) => (
  <Card title={circuitName}>
    {/* <CardTitle>{circuitName}</CardTitle>*/}
    <div className="content">
      <Flag country={country} />
      <h3>{circuitName}</h3>
      <p>
        {locality}, {country}
      </p>
      <div>
        <b>Round:</b> {round}
      </div>
      <div>
        <b>Race:</b> {raceName}
      </div>
      <div>
        <b>Date:</b> {date}
      </div>
    </div>
    <CardButton onClick={onSeasonSelect} title="Go to standings">
      Go to standings &rarr;
    </CardButton>
  </Card>
);

export default SeasonCard;

export const Card = styled.div`
  width: 100%;
  margin-bottom: ${({ theme }) => theme.grid * 2}px;
  background: white;
  color: ${({ theme }) => theme.gray1};
  box-shadow: 0 0 10px 1px rgba(0, 0, 0, 0.3);

  @media (min-width: 600px) {
    margin: 8px;
    width: 47%;
  }

  @media (min-width: 900px) {
    width: 31%;
  }

  @media (min-width: 1200px) {
    width: 22%;
  }

  @media (min-width: 1500px) {
    width: 18%;
  }

  .content {
    padding: ${({ theme }) => theme.grid * 2.5}px;
    background: ${({ theme }) => theme.gray2};
  }

  h4 {
    text-align: center;
  }
`;
