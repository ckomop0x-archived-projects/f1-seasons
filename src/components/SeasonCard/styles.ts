import styled from 'styled-components';

import { FlagWrapper } from 'components/Flag/FlagStyles';

export const Round = styled.div`
  background-color: ${({ theme }) => theme.gray1};
  height: 100%;
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 16px;
  font-size: 20px;
`;

export const RoundTitle = styled.div`
  color: ${({ theme }) => theme.gray2};
`;

export const RoundNumber = styled.div`
  padding-left: 16px;
  color: ${({ theme }) => theme.gray2};
`;

export const CircuitName = styled.div`
  padding: 8px 0;
  h3 {
    margin: 0;
  }
  span {
    opacity: 0.8;
  }
`;

export const RaceName = styled.div`
  padding-left: 16px;
  width: 100%;
  display: flex;
  justify-content: flex-start;

  b {
    width: 64px;
    padding-right: 20px;
  }
`;

export const DetailsButton = styled.button`
  margin-top: 16px;
  width: 100%;
  border: 0;
  padding: 16px;
  color: ${({ theme }) => theme.gray2};
  background-color: ${({ theme }) => theme.primaryRed};
`;

export const SeasonCardWrapper = styled.div`
  border: 1px solid ${({ theme }) => theme.gray1};
  background-color: ${({ theme }) => theme.gray2};
  border-radius: 4px;
  width: 100%;
  color: ${({ theme }) => theme.gray1};
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-bottom: 20px;
`;

export const CountryDetails = styled.div`
  padding: 4px 0 0 12px;
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;

  ${FlagWrapper} {
    padding-top: 6px;
    padding-right: 20px;
    border-radius: 0;

    img {
      width: 48px;
      border-radius: 0;
    }
  }
`;
