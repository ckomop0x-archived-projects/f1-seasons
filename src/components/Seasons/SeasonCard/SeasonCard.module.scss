.wrapper {
  display: flex;
  flex-direction: column;
  box-sizing: border-box;
  width: 100%;
  margin-bottom:  calc(var(--grid) * 2);
  background: var(--color-gray2);
  color: var(--color-gray1);
  box-shadow: 0 0 10px 1px rgba(0, 0, 0, 0.3);
  padding: calc(var(--grid) * 2);
  min-height: calc(var(--grid) * 40);

  @media (min-width: 600px) {
    margin: 8px;
    flex-basis: 47%;
  }

  @media (min-width: 900px) {
    flex-basis: 31%;
  }

  @media (min-width: 1200px) {
    flex-basis: 22%;
  }

  @media (min-width: 1500px) {
    flex-basis: 18%;
  }

  .content {
    display: flex;
    flex-direction: row;
    align-self: flex-start;
    margin-bottom: calc(var(--grid) * 3);
  }

  h3 {
    font-size: 18px;
    margin: 0 0 0 20px;
  }

  h4 {
    text-align: center;
    font-size: 18px;
  }
}

.raceInfo {
  padding: var(--grid);
}
