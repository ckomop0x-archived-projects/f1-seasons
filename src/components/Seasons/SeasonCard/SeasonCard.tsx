import SeasonCardButton from 'components/Seasons/SeasonCard/SeasonCardButton/SeasonCardButton';
import { FC } from 'react';

import Flag from '../../UI/Flag';

import styles from './SeasonCard.module.scss';

export interface SeasonCardProps {
  raceName: string;
  circuitName: string;
  season: string;
  round: string;
  country: string;
  locality: string;
  circuitId: string;
  date: string;
  time: string;
}

const SeasonCard: FC<SeasonCardProps> = ({
  raceName,
  circuitName,
  season,
  round,
  country,
  locality,
  date,
}) => (
  <div title={circuitName} className={styles.wrapper}>
    <div className={styles.content}>
      <Flag country={country} />
      <h3>{circuitName}</h3>
    </div>
    <div className={styles.raceInfo}>
      <div>
        <b>City:</b> {locality}, {country}
      </div>
      <div>
        <b>Round:</b> {round}
      </div>
      <div>
        <b>Race:</b> {raceName}
      </div>
      <div>
        <b>Date:</b> {date}
      </div>
    </div>

    <SeasonCardButton link={`/${season}/${round}`} title="Go to standings">
      Go to standings &rarr;
    </SeasonCardButton>
  </div>
);

export default SeasonCard;
