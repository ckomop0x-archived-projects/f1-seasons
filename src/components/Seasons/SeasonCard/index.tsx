import * as React from 'react';
import { FC } from 'react';

import CardButton from '../../../styles/CardButton';
import { CardTitle } from '../../../styles/CardTitle';
import Flag from '../../Flag';

import { Card } from './SeasonCardStyles';

export interface SeasonCardProps {
  raceName: string;
  circuitName: string;
  season: string;
  round: string;
  country: string;
  locality: string;
  circuitId: string;
  date: string;
  time: string;
}

const SeasonCard: FC<SeasonCardProps> = ({
  raceName,
  circuitName,
  season,
  round,
  country,
  locality,
  date,
}) => (
  <Card title={circuitName}>
    <CardTitle>{circuitName}</CardTitle>
    <div className="content">
      <Flag country={country} />
      <h4>
        {locality}, {country}
      </h4>
      <div>
        <b>Round:</b> {round}
      </div>
      <div>
        <b>Race:</b> {raceName}
      </div>
      <div>
        <b>Date:</b> {date}
      </div>
    </div>
    <CardButton href={`/${season}/${round}`} title="Go to standings">
      Go to standings &rarr;
    </CardButton>
  </Card>
);

export default SeasonCard;
