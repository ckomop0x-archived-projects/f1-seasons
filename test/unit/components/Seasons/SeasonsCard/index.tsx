import * as React from 'react';
import * as renderer from 'react-test-renderer';
import SeasonCard from '../../../../../src/components/SeasonCard';

it('renders the heading', () => {
  const props = {
    raceName: 'Race',
    circuitName: 'Circuit',
    season: 'season',
    round: 'round',
    country: 'country',
    locality: 'locality',
    circuitId: 'circuitId',
    date: 'date',
    time: 'time',
    onSeasonSelect: jest.fn()
  };
  const wrapper = renderer.create(<SeasonCard {...props} />).toJSON();

  expect(wrapper).toMatchSnapshot();
});
