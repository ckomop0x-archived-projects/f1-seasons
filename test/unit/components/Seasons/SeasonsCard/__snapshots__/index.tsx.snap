// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the heading 1`] = `
<div
  className="sc-bwzfXH jDXDDZ"
>
  <div
    className="title"
  >
    Round: 
    round
  </div>
  <div
    className="content"
  >
    <h3>
      Circuit
    </h3>
    <h4>
      locality
      , 
      country
    </h4>
    <div>
      Race – 
      Race
      ,
    </div>
    <div>
      Date – 
      date
    </div>
    <button
      className="sc-bdVaJa jYJGyW"
      onClick={[MockFunction]}
      title="Read more on Wiki"
    >
      Go to season standings
    </button>
  </div>
</div>
`;
